src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let sep;
+
+    // in case the passed separator is null
+    if (separator === null) {
+      sep = 'null';
+    } else {
+      sep = separator;
+    }
+
+    // joining the array excluding the last element
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
i suppose a generic check would do the job?
```suggestion
      if (this[i]) { ...
```

Reply:
I tried this, the case where an element is 0 is the problem.